### PR TITLE
Renames the opencensus-monitoring directory and adds comments to it.

### PR DESCRIPTION
**Pull request by imccarten1**

### CODE REVIEW

File: opentelemetry_collector/receiver/vmimageagereceiver/factory.go [hunk 1]
```diff
@@ -35,6 +39,8 @@ func (f *Factory) CreateDefaultConfig() configmodels.Receiver {
 	}
 }
 
+// CreateTraceReceiver generates an error because this receiver does not
```
Comment (yanweiguo) [inline opentelemetry_collector/receiver/vmimageagereceiver/factory.go]: Is CreateTraceReceiver necessary if it always return an error?

File: opentelemetry_collector/receiver/metricgenerator/metric_generator.go [hunk 2]
```diff
@@ -29,6 +30,13 @@ func MakeInt64TimeSeries(val int64, startTime, now time.Time, labels []*metricsp
 	}
 }
 
+// MakeExponentialBucketOptions generates a proto representation of a config which,
+// defines a distribution's bounds. This defines maxExponent + 1 (= N) buckets. The boundaries for bucket
```
Comment (yanweiguo) [inline opentelemetry_collector/receiver/metricgenerator/metric_generator.go]: Not sure whether `maxExponent + 1 (= N)` is the right expression. It confuses me at first. Do we need to introduce `N`? How about 

```
This defines maxExponent + 1 buckets. ....
...
[boundsBase ^ (i - 1)], boundsBase ^ i) for 0 < i < maxExponent
[boundsBase ^ i, +infinity) for i == maxExponent
```

File: opentelemetry_collector/receiver/vmimageagereceiver/factory.go (hunk 1 above)
Comment (imccarten1) [reply opentelemetry_collector/receiver/vmimageagereceiver/factory.go]: It's required to match the receiver Factory interface. https://github.com/open-telemetry/opentelemetry-collector/blob/4da23d11f0d8e3fa3309ac3804c9421b95235f32/receiver/factory.go#L63

File: opentelemetry_collector/receiver/metricgenerator/metric_generator.go (hunk 2 above)
Comment (imccarten1) [reply opentelemetry_collector/receiver/metricgenerator/metric_generator.go]: I've removed the reference to N and also fixed an error in comment's explanation.

**Pull request merged**